src/main/java/org/opentripplanner/routing/request/RoutingRequest.java

```diff
@@ -820,7 +820,12 @@ public void setBikeBoardCost(int bikeBoardCost) {
     public void setPreferredAgencies(String s) {
         if (!s.isEmpty()) {
             preferredAgencies = new HashSet<>();
-            Collections.addAll(preferredAgencies, s.split(","));
+            Collections.addAll(
+                preferredAgencies,
+                Arrays.stream(s.split(","))
+                    .map(FeedScopedId::convertFromString)
+                    .toArray(FeedScopedId[]::new)
+            );
```

Comment:
This is repeated several times.  I suggest making a static parse function in the FeedScopedId.

Reply:
Added a static method `convertFromConcatenatedString ` in `FeedScopedId` for this.